src/shared/Microsoft.Git.CredentialManager.Tests/ConfigurationServiceTests.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+using System.Threading.Tasks;
+using Microsoft.Git.CredentialManager.Tests.Objects;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Git.CredentialManager.Tests
+{
+    public class ConfigurationServiceTests
+    {
+        [Fact]
+        public async Task ConfigurationService_ConfigureAsync_System_ComponentsAreConfiguredWithSystem()
+        {
+            var context = new TestCommandContext();
+            var service = new ConfigurationService(context);
+
+            var component1 = new Mock<IConfigurableComponent>();
+            var component2 = new Mock<IConfigurableComponent>();
+            var component3 = new Mock<IConfigurableComponent>();
+
+            service.AddComponent(component1.Object);
+            service.AddComponent(component2.Object);
+            service.AddComponent(component3.Object);
+
+            await service.ConfigureAsync(true);
+
+            component1.Verify(x => x.ConfigureAsync(true, context.Environment, It.IsAny<IGitConfiguration>()), Times.Once);
+            component2.Verify(x => x.ConfigureAsync(true, context.Environment, It.IsAny<IGitConfiguration>()), Times.Once);
+            component3.Verify(x => x.ConfigureAsync(true, context.Environment, It.IsAny<IGitConfiguration>()), Times.Once);
+        }
+
+        [Fact]
+        public async Task ConfigurationService_ConfigureAsync_User_ComponentsAreConfiguredWithUser()
+        {
+            var context = new TestCommandContext();
+            var service = new ConfigurationService(context);
+
+            var component1 = new Mock<IConfigurableComponent>();
+            var component2 = new Mock<IConfigurableComponent>();
+            var component3 = new Mock<IConfigurableComponent>();
+
+            service.AddComponent(component1.Object);
+            service.AddComponent(component2.Object);
+            service.AddComponent(component3.Object);
+
+            await service.ConfigureAsync(false);
+
+            component1.Verify(x => x.ConfigureAsync(false, context.Environment, It.IsAny<IGitConfiguration>()), Times.Once);
+            component2.Verify(x => x.ConfigureAsync(false, context.Environment, It.IsAny<IGitConfiguration>()), Times.Once);
+            component3.Verify(x => x.ConfigureAsync(false, context.Environment, It.IsAny<IGitConfiguration>()), Times.Once);
+        }
+
+        [Fact]
```

Comment:
The change between this / the next test seems to be based on weather or not we run Async.  Can you confirm how that relates to the change in test name (System/User)

Reply:
> or not we run Async.

This isn't correct. We're always doing `await service.UnconfigureAsync(bool)` but the boolean we pass is 'unconfigure the user' or 'unconfigure the system', hence the test name change. 😄

Reply:
I see, that is a bit confusing when just given a true/false, but makes sense

Reply:
+1 maybe consider an enum for the level to configure for clarity?

Reply:
Added and plumbed through an enum!